src/lender/adapters/mkr/clerk.sol

```diff
@@ -224,9 +236,37 @@ contract Clerk is Auth, Math {
         creditline = safeSub(creditline, amountDAI);
     }
 
+    function heal(uint amountDAI) public auth {    
+        uint collatDeficitDAI = collatDeficit();
+        require(collatDeficitDAI > 0, "no healing required");
+    
+        // heal max up to the required missing collateral amount
+        if (collatDeficitDAI < amountDAI) {
+            amountDAI = collatDeficitDAI;
+        }
+  
+        require((validate(0, amountDAI, 0, 0) == 0), "supply not possible, pool constraints violated");
+        // mint drop and move into cdp
+        uint priceDROP = assessor.calcSeniorTokenPrice();
+        uint collateralDROP = rdiv(amountDAI, priceDROP);
+        tranche.mint(address(this), collateralDROP);
+        collateral.approve(address(mgr), collateralDROP);
+        mgr.join(collateralDROP);
+        // increase seniorAsset by amountDAI
+        updateSeniorAsset(0, amountDAI);
+    }
+
+    // heal the cdp and put in more drop in case the collateral value has fallen below the bufferedmat ratio
+    function heal() public auth {
+        uint collatDeficitDAI = collatDeficit();
+        if (collatDeficitDAI > 0) {
+            heal(collatDeficitDAI);
+        }
+    }
+
     // checks if the Maker credit line increase could violate the pool constraints // -> make function pure and call with current pool values approxNav
-    function validate(uint juniorSupplyDAI, uint juniorRedeemDAI, uint seniorSupplyDAI, uint seniorRedeemDAI) internal {
-        require((coordinator.validate(juniorSupplyDAI, juniorRedeemDAI, seniorSupplyDAI, seniorRedeemDAI) == 0), "supply not possible, pool constraints violated");
+    function validate(uint juniorSupplyDAI, uint juniorRedeemDAI, uint seniorSupplyDAI, uint seniorRedeemDAI) internal returns (int) {
+        return coordinator.validate(juniorSupplyDAI, juniorRedeemDAI, seniorSupplyDAI, seniorRedeemDAI);
```

Comment:
I think this has an impact on every validate call in the clerk.

Like `raise` etc.

We need to check for equal zero at least in the raise method.

We should also add a constant for that.